### PR TITLE
Feature/issue 197: [Admin page] Front-end - Back-end integration

### PR DESCRIPTION
**Pull request by milrusy**
## Github ticket

* [Github ticket](https://github.com/ita-social-projects/VictoryCenter-Back/issues/197)

## Description

Call the required endpoints using Axios methods; the final result should exactly match the front-end mocks.

## How it looks

image

## Summary of change

Implemented front-end integration with the back-end for the admin page using the authenticated API client. Proper data fetching logic was added to ensure accurate loading of team members and synchronization with the server state.

## How to recreate changes

⚠️Note that the backend branch with admin authorization has not been merged yet. To run the application properly, make sure to checkout and use branch Feature/issue-41 on the backend side.
1. Start the VictoryCenter-Back server at https://localhost:5001/.
2. In the VictoryCenter-Client project, update the REACT_APP_BACKEND_URL variable in the .env file to https://localhost:5001/api/.
3. Start the VictoryCenter-Client server.
4. Open admin team page in browser: http://localhost:3000/admin-page/team

## CHECK LIST
- [ ]  New tests was added or existing was modified
- [ ]  Changes has severe impact on users
- [ ]  Changes has medium impact on users
- [ ]  Changes has light impact on users
- [ ]  Test covetage was updated
- [ ]  PR meets all conventions

## Summary by CodeRabbit

* **New Features**
  * Added full backend integration for team member management, including creating, editing, deleting, saving as draft, publishing, and reordering members from the admin interface.
  * Introduced bidirectional mapping between team category names and their numeric codes.
  * Enhanced UI to support real-time updates and refreshes after member actions.
  * Added comprehensive API service and data transfer object definitions for team members.
  * Improved form handling in the add member modal with controlled form state and validation for draft saving.

* **Bug Fixes**
  * Cleaned up Jest configuration to remove duplicated entries.

* **Tests**
  * Expanded and improved test coverage for team member API operations and admin UI interactions, including draft-saving and member creation workflows.
  * Added mocks for admin context HTTP client in tests to better simulate API interactions.
  * Enhanced test simulations of form state changes for more accurate coverage.

* **Documentation**
  * Updated project directory structure in documentation to reflect new data-fetching services.

### CODE REVIEW

File: src/pages/admin/team/components/team-page-content/TeamPageContent.tsx [hunk 1]
```diff
@@ -19,17 +25,30 @@ export const TeamPageContent = () => {
...
 
+    const handleAddMember = async (member: MemberFormValues) => {
+        await TeamMembersApi.postPublished(client, member);
+        setRefetchKey(prev => prev+1);
+    }
+
+    const handleSaveDraft = async (member: MemberFormValues) => {
+        await TeamMembersApi.postDraft(client, member);
+        setRefetchKey(prev => prev+1);
+    }
```
[the remark this answers is not in the document]
Comment (MarkBevz50) [reply src/pages/admin/team/components/team-page-content/TeamPageContent.tsx:36]: My suggestions:
Implement the suggested try/catch blocks for error handling.
Add loading states to improve UX during API calls.

File: src/services/data-fetch/admin-page-data-fetch/team-page-data-fetch/TeamMembersApi.test.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
+import { AxiosInstance, AxiosResponse } from "axios";
+import { MemberFormValues } from "../../../../pages/admin/team/components/member-form/MemberForm";
```
Comment (MarkBevz50) [inline src/services/data-fetch/admin-page-data-fetch/team-page-data-fetch/TeamMembersApi.test.ts:2]: My suggestions:
Add tests for error cases (e.g., 404, 500 responses).
Verify edge cases (e.g., empty mockDto).

File: src/services/data-fetch/admin-page-data-fetch/team-page-data-fetch/TeamMembersApi.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import { Member } from '../../../../pages/admin/team/components/members-list/MembersList';
```
Comment (MarkBevz50) [inline src/services/data-fetch/admin-page-data-fetch/team-page-data-fetch/TeamMembersApi.ts:1]: My suggestion:
Add error handling with try/catch for each method.

**Review by MarkBevz50 — APPROVED** [review]
So, I'll approve this, but I recommend making some requested changes. Overall, it looks good to me!

File: src/services/data-fetch/admin-page-data-fetch/team-page-data-fetch/TeamMembersApi.ts (hunk 3 above)
Comment (Rominos7) [reply src/services/data-fetch/admin-page-data-fetch/team-page-data-fetch/TeamMembersApi.ts:1]: Not necessary 
It's better to do in places where you actually calling those methods 
Because if you do try/catch here it will encapsulate error within itself and it would be harder to handle it.

**Review by Rominos7 — APPROVED** [review]
